src/client/scripts/esm/chess/variants/variant.ts

```diff
@@ -55,21 +55,20 @@ const gameruleModificationsOfOmegaShowcasings = { promotionRanks: null, moveRule
 // 	}
 // }
 
-// Not needed since it's imported
-/*interface GameRules {
+interface GameRuleModifications {
 	promotionRanks?: (number | null)[] | null,
 	moveRule?: number | null,
 	turnOrder?: string[],
 	promotionsAllowed?: {
-		white: string[],
-		black: string[]
+		white?: string[],
+		black?: string[]
```

Comment:
In the future, we may have more colors. Lets just define promotionsAllowed as `Object<string, string[]>`